middleware/src/system/system.go

```diff
@@ -0,0 +1,53 @@
+//package systemenv provides functionality to get data such as open ports and services running from the system the middleware is deployed on
+package environment
+
+// SystemEnv provides some information on the system we are running on.
+type SystemEnv struct {
+	network                                            string
+	electrsRPCPort                                     string
+	bitcoinRPCUser, bitcoinRPCPassword, bitcoinRPCPort string
+	lightningRPCPath                                   string
+}
+
+// NewSystemEnv returns a new SystemEnv instance.
+func NewSystemEnv(bitcoinRPCUser, bitcoinRPCPassword, bitcoinRPCPort, lightningRPCPath, electrsRPCPort, network string) SystemEnv {
+	systemEnv := SystemEnv{
+		bitcoinRPCUser:     bitcoinRPCUser,
+		bitcoinRPCPassword: bitcoinRPCPassword,
+		bitcoinRPCPort:     bitcoinRPCPort,
+		lightningRPCPath:   lightningRPCPath,
+		electrsRPCPort:     electrsRPCPort,
+		network:            network,
+	}
+	return systemEnv
+}
+
+// GetBitcoinRPCUser is a getter for bitcoinRPCUser
+func (systemEnv *SystemEnv) GetBitcoinRPCUser() string {
```

Comment:
These getter methods have no further logic, so I'd suggest removing them and letting the caller just access the fields they want as appropriate. If you follow my suggestion above and make the struct part of the `middleware` package you can have all struct fields unexported (lower case), which prevents any external caller from mutating the values.

Reply:
I find it weird to expose struct fields that are read-only, except if changes in the system environment occur to a base package like the middleware in this case. If any of these fields need json marshalling, or actually allow writing from the outside, I will remove the getter and expose the struct field instead.

Reply:
If we do keep this struct in a separate package and don't want to export the fields, yes, we probably need getters.